code/modules/research/xenobiology/crossbreeding/reproductive.dm

```diff
@@ -3,47 +3,62 @@ Reproductive extracts:
 	effect_desc = "When fed monkey cubes it produces more extracts. Bio bag compatible as well."
 	var/extract_type = /obj/item/slime_extract/
-	var/cubes_eaten = 0
+	var/cooldown = 3 SECONDS
 	var/last_produce = 0
-	var/cooldown = 30 // 3 seconds.
+	var/datum/component/storage/concrete/extract_inventory/slimeStorage
+	
+/obj/item/slimecross/reproductive/examine()
+	. = ..()
+	. += span_danger("It appears to have eaten [length(contents)] Monkey Cube[p_s()]")
+
+/obj/item/slimecross/reproductive/Initialize()
+	. = ..()
+	slimeStorage = AddComponent(/datum/component/storage/concrete/extract_inventory)
 
 /obj/item/slimecross/reproductive/attackby(obj/item/O, mob/user)
 	if((last_produce + cooldown) > world.time)
 		to_chat(user, span_warning("[src] is still digesting!"))
 		return
+
+	if(length(contents) >= 3) //if for some reason the contents are full, but it didnt digest, attempt to digest again
+		to_chat(user, span_warning("[src] appears to be full but is not digesting! Maybe poking it stimulated it to digest."))
+		slimeStorage.processCubes(src, user)
+		return
+
 	if(istype(O, /obj/item/storage/bag/bio))
 		var/list/inserted = list()
-		SEND_SIGNAL(O, COMSIG_TRY_STORAGE_TAKE_TYPE, /obj/item/food/monkeycube, src, 1, null, null, user, inserted)
+		SEND_SIGNAL(O, COMSIG_TRY_STORAGE_TAKE_TYPE, /obj/item/food/monkeycube, src, 3 - length(contents), null, null, user, inserted)
 		if(inserted.len)
-			var/obj/item/food/monkeycube/M = inserted[1]
-			if(istype(M))
-				eat_cube(M, user)
+			to_chat(user, span_notice("You feed [length(inserted)] Monkey Cube[p_s()] to [src], and it pulses gently."))
+			playsound(src, 'sound/items/eatfood.ogg', 20, TRUE)
+			slimeStorage.processCubes(src, user)
 		else
 			to_chat(user, span_warning("There are no monkey cubes in the bio bag!"))
-	if(istype(O,/obj/item/food/monkeycube))
-		eat_cube(O, user)
-	if(cubes_eaten >= 3)
-		var/cores = rand(1,4)
-		visible_message(span_notice("[src] briefly swells to a massive size, and expels [cores] extract[cores > 1 ? "s":""]!"))
-		playsound(src, 'sound/effects/splat.ogg', 40, TRUE)
-		last_produce = world.time
-		for(var/i = 0, i < cores, i++)
-			new extract_type(get_turf(loc))
-		cubes_eaten = 0
-
-/obj/item/slimecross/reproductive/proc/eat_cube(obj/item/food/monkeycube, mob/user)
-		qdel(monkeycube)
-		cubes_eaten++
-		to_chat(user, span_notice("You feed [monkeycube] to [src], and it pulses gently."))
-		playsound(src, 'sound/items/eatfood.ogg', 20, TRUE)
+		return
 
+	else if(istype(O, /obj/item/food/monkeycube))
+		if(length(contents) < 3)
+			if(SEND_SIGNAL(src, COMSIG_TRY_STORAGE_INSERT, O, user, TRUE, TRUE))
+				to_chat(user, span_notice("You feed 1 Monkey Cube to [src], and it pulses gently."))
+				slimeStorage.processCubes(src, user)
+				playsound(src, 'sound/items/eatfood.ogg', 20, TRUE)
+				return
+			else
+				to_chat(user, span_notice("The [src] rejects the Monkey Cube!")) //in case it fails to insert for whatever reason you get feedback
```

Comment:
```suggestion
				to_chat(user, span_notice("The [src] rejects the Monkey Cube!")) //in case it fails to insert for whatever reason you get feedback
				
/obj/item/slimecross/reproductive/Destroy()
	slimeStorage = null
	return ..()

```
removes the component on destroy